ddR/R/ddR.R

```diff
@@ -55,20 +56,19 @@ ddR.env$nexecutors <- 1
 #' library(distributedR.ddR); useBackend(distributedR)
 #' } 
 #' @export
-useBackend <- function(driver, ...) {
+useBackend <- function(driver = parallel, ...) {
 
-  if(!is.null(ddR.env$driver)) shutdown(ddR.env$driver)
+    # validate first or else shutdown can fail on subsequent calls
+    if(!is(driver, "ddRDriver")){
+        stop("Invalid driver object specified")
+    }
 
-  if(!extends(class(driver)[[1]],"ddRDriver")) stop("Invalid driver object specified")
+    if(!is.null(ddR.env$driver)) shutdown(ddR.env$driver)
 
-  if(!extends(driver@DListClass,"DObject")) stop("The driver DList class does not extend ddR::Dobject")
```

Comment:
Has a need for this come up in the past?

This approach is simpler, and the only way it can fail is if people are writing their own drivers. In this case they probably expect to encounter problems, and should only be hopeful that it will work when it passes the new `test-driver-common.R` file. Even when ddR is fully mature my guess is there will be ~5 robust drivers that people use. 

There are many possible checks that could be done to make sure the driver _actually_ works- I would think that checking for the definition of `do_mapply` and `do_collect` would be the most important.